lib/model/rwfolder.go

```diff
@@ -1378,31 +1425,14 @@ func (f *sendReceiveFolder) performFinish(state *sharedPullerState) error {
 			fallthrough
 		// Permissions have changed, means the file has changed, rescan.
 		case !f.ignorePermissions(state.curFile) && state.curFile.HasPermissionBits() && !scanner.PermsEqual(state.curFile.Permissions, curMode):
-			l.Debugln("file modified but not rescanned; not finishing:", state.curFile.Name)
-			// Scan() is synchronous (i.e. blocks until the scan is
-			// completed and returns an error), but a scan can't happen
-			// while we're in the puller routine. Request the scan in the
-			// background and it'll be handled when the current pulling
-			// sweep is complete. As we do retries, we'll queue the scan
-			// for this file up to ten times, but the last nine of those
-			// scans will be cheap...
-			go f.Scan([]string{state.curFile.Name})
+			l.Debugln("file unknow or modified but not scanned; not finishing:", state.curFile.Name)
+			toBeScanned[state.curFile.Name] = struct{}{}
 			return fmt.Errorf("file modified but not rescanned; will try again later")
 		}
 
 		switch {
 		case stat.IsDir() || stat.IsSymlink():
-			// It's a directory or a symlink. These are not versioned or
-			// archived for conflicts, only removed (which of course fails for
-			// non-empty directories).
-
-			// TODO: This is the place where we want to remove temporary files
-			// and future hard ignores before attempting a directory delete.
-			// Should share code with f.deletDir().
-
-			if err = osutil.InWritableDir(f.fs.Remove, f.fs, state.file.Name); err != nil {
-				return err
-			}
+			panic("Symlinks and directories should be handled directly")
```

Comment:
This can cause a panic if I replace a file with a directory as it's being downloaded.
I think this should be a conflict, or do what it does now.

Reply:
I misread this, will have a second look at it. the todo comment is confusing

Reply:
I misread this, will have a second look at it. the todo comment is confusing